stdlib/test/builtin/test_string_literal.mojo

```diff
@@ -142,6 +142,43 @@ fn test_repr() raises:
     assert_equal(StringLiteral.__repr__("\x7f"), r"'\x7f'")
 
 
+def test_format_simple():
+    assert_true(
+        "{first} {second}".format_simple(first="Hello", second="world")
+        == "Hello world"
+    )
+
+    assert_true(
+        "Number: {n}".format_simple(
+            n=123,
+        )
+        == "Number: 123"
+    )
+
+    with assert_raises(contains="A replacement value contains {}"):
+        _ = "Should raise simple error: {n}".format_simple(n="{123}")
+
+    assert_true(
```

Comment:
Use `assert_equals` so if a test fails it will print the two inputs and we can easily see which test failed and how.

Reply:
Yes, it is an excellent idea, need help for this one, as it already raises in the `with` block

Reply:
@rd4com can you clarify what you need help with?

Reply:
Thanks :+1: , how to both `assert_raise` and `assert_equal` (because there is no value if it raises)

Reply:
It's one or the other; if it raises, you can't compare it's output. And if it had an output, it did not raise, so you can use `assert_equal`. Maybe I'm misunderstanding, do you have a concrete example?

Reply:
I think i'm confused, it is expected that it raises in the next example:
```mojo
with assert_raises(contains="args[1] was provided but not used"):
     print("First {0}".format("First", True))
```
No value is returned, not sure how to ```assert_equal```;

Maybe @bgreni was refering to an ```assert_true``` below an ```assert_raise``` ?

Reply:
Yes, he was referring to the assert_true, and I think you already changed that.